python/lib/core/dmod/core/dataset.py

```diff
@@ -117,7 +117,19 @@ class Config:
         # TODO: in future deprecate setting properties unless through a setter method
         validate_assignment = True
         arbitrary_types_allowed = True
-        field_serializers = {"uuid": lambda f: str(f)}
+
+        def _serialize_datetime(self: "Dataset", value: Optional[datetime]) -> Optional[str]:
+            if isinstance(value, datetime):
+                return value.strftime(self._SERIAL_DATETIME_STR_FORMAT)
+            return value
+
+        field_serializers = {
+            "uuid": str,
```

Comment:
I think `uuid` should have something akin to the lambda used for `manager_uuid`, although perhaps it's declaration needs adjustment.  That is:  

``` Python
uuid: Optional[UUID] = Field(default_factory=uuid4)
```

If I follow correctly, this implies the possibility of explicitly receiving ``None``.  It's possible there's a subtlety about Pydantic I'm missing that _requires_ this to be declared as _Optional_ even though it really isn't (although if that's the case, it's bothersome).